src/xunit.netcore.extensions/Discoverers/ConditionalFactDiscoverer.cs

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Xunit.Abstractions;
+using Xunit.Sdk;
+
+namespace Xunit.NetCore.Extensions
+{
+    public class ConditionalFactDiscoverer : FactDiscoverer
+    {
+        private readonly IMessageSink _diagnosticMessageSink;
+
+        public ConditionalFactDiscoverer(IMessageSink diagnosticMessageSink) : base(diagnosticMessageSink)
+        {
+            _diagnosticMessageSink = diagnosticMessageSink;
+        }
+
+        public override IEnumerable<IXunitTestCase> Discover(
+            ITestFrameworkDiscoveryOptions discoveryOptions, ITestMethod testMethod, IAttributeInfo factAttribute)
+        {
+            MethodInfo testMethodInfo = testMethod.Method.ToRuntimeMethod();
+
+            string conditionMemberName = factAttribute.GetConstructorArguments().FirstOrDefault() as string;
+            MethodInfo conditionMethodInfo;
+            if (conditionMemberName == null ||
+                (conditionMethodInfo = LookupConditionalMethod(testMethodInfo.DeclaringType, conditionMemberName)) == null)
+            {
+                return new[] {
+                    new ExecutionErrorTestCase(
+                        _diagnosticMessageSink,
+                        discoveryOptions.MethodDisplayOrDefault(),
+                        testMethod,
+                        "An appropriate member \"" + conditionMemberName + "\" could not be found.")
```

Comment:
Can we explain "appropriate"?  It looks like it has to be a static 0-argument method (or get-accessor), I'm not sure if it has visibility requirements, or not.